src/aics-image-viewer/state/subscribers.ts

```diff
@@ -0,0 +1,179 @@
+      // if slider was same as active axis view mode:  [viewerSettings.slice[axis], viewerSettings.slice[axis] + 1.0/volumeSize[axis]]
+      // if in 3d mode: viewerSettings.region[axis]
+      // else: [0,1]
+      let isOrthoAxis = false;
+      let axismin = 0.0;
+      let axismax = 1.0;
+      if (viewMode === ViewMode.threeD) {
+        axismin = minval;
+        axismax = maxval;
+      } else {
+        isOrthoAxis = activeAxisMap[viewMode] === axis;
+        const oneSlice = 1 / image.imageInfo.volumeSize[axis];
+        axismin = isOrthoAxis ? slice : 0.0;
+        axismax = isOrthoAxis ? slice + oneSlice : 1.0;
+        if (axis === "z" && viewMode === ViewMode.xy) {
+          view3d.setZSlice(image, Math.floor(slice * image.imageInfo.volumeSize.z));
+        }
+      }
+
+      // view3d wants the coordinates in the -0.5 to 0.5 range
+      view3d.setAxisClip(image, axis, axismin - 0.5, axismax - 0.5, isOrthoAxis);
+      // TODO necessary?
+      // view3d.setCameraMode(viewMode);
+      // TODO under some circumstances, this effect will trigger a load. Ideally, this would be reflected in the load
+      //   state managed by `useVolume`. This is complicated by the fact that the relevant methods (`setAxisClip` and
+      //   `setZSlice`) don't provide a channel load callback like other load-triggering methods (e.g. `setTime`).
+    };
+  };
+
+  const unsubscribers = [
+    // show bounding box
+    store.subscribe(
+      select("showBoundingBox"),
+      (showBoundingBox) => view3d.setShowBoundingBox(image, showBoundingBox),
+      REF_EQ
+    ),
+
+    // bounding box color
+    store.subscribe(
+      select("boundingBoxColor"),
+      (boundingBoxColor) => view3d.setBoundingBoxColor(image, boundingBoxColor),
+      REF_EQ
+    ),
+
+    // render mode
+    store.subscribe(
+      select("renderMode"),
+      (renderMode) => {
+        view3d.setMaxProjectMode(image, renderMode === RenderMode.maxProject);
+        view3d.setVolumeRenderMode(renderMode === RenderMode.pathTrace ? RENDERMODE_PATHTRACE : RENDERMODE_RAYMARCH);
+        view3d.updateActiveChannels(image);
+      },
+      REF_EQ
+    ),
+
+    // mask alpha
+    store.subscribe(
+      select("maskAlpha"),
+      (maskAlpha) => {
+        view3d.updateMaskAlpha(image, alphaSliderToImageValue(maskAlpha));
+        view3d.updateActiveChannels(image);
+      },
+      REF_EQ
+    ),
+
+    // density
+    store.subscribe(
+      select("density"),
+      (density) => view3d.updateDensity(image, densitySliderToImageValue(density)),
+      REF_EQ
+    ),
+
+    // gamma
+    store.subscribe(
+      select("levels"),
+      (levels) => {
+        const { min, max, scale } = gammaSliderToImageValues(levels);
+        view3d.setGamma(image, min, max, scale);
+      },
+      REF_EQ
+    ),
+
+    // interpolation
+    store.subscribe(
+      select("interpolationEnabled"),
+      (enabled) => view3d.setInterpolationEnabled(image, enabled),
+      REF_EQ
+    ),
+
```

Comment:
Maybe easy question, do all of these values get assigned on `view3d` from state at initialization?